src/GitVersionCore/Configuration/BranchConfigurationCalculator.cs

```diff
@@ -214,7 +215,7 @@ private static BranchConfig ChooseMasterOrDevelopIncrementStrategyIfTheChosenBra
             BranchConfig masterOrDevelopConfig = null;
             var developBranchRegex = config.Branches[Config.DevelopBranchKey].Regex;
             var masterBranchRegex = config.Branches[Config.MasterBranchKey].Regex;
-            if (Regex.IsMatch(chosenBranch.FriendlyName, developBranchRegex, RegexOptions.IgnoreCase))
+            if (Regex.IsMatch(chosenBranch.Name.Friendly, developBranchRegex, RegexOptions.IgnoreCase))
```

Comment:
Can we expose this as `IBranch.IsDevelop` in a future PR?

```suggestion
            if (chosenBranch.IsDevelop)
```

Reply:
Yes I think that could be a good addition, both `IsDevelop` and `IsMain`. Does this mean we need to access the config in the branch class for the regex? What could be the internal implementation in that case

Reply:
Yes, the configuration can probably be exposed through an interface that is injected into the `Branch` class constructor?